python/cudf/udf_cpp/shim.cu

```diff
@@ -720,3 +722,49 @@ make_definition_corr(BlockCorr, int64, int64_t);
 
 #undef make_definition_corr
 }
+
+/*
+NRT CUDA functions
+*/
+
+// Only used to allocate the right amount of space, see below
+struct meminfo_and_str {
+  NRT_MemInfo mi;
+  udf_string st;
+};
+
+/*
+Create a new MemInfo object holding the reference count of a udf_string. When returning
+new strings, shim functions expect a pointer to a stack allocated buffer into which it
+will construct the udf_string it returns. Since one can not safely build a MemInfo object
+around this stack memory, we store a copy of the udf_string itself next to the MemInfo
+which manages the lifetime of the udf_string.
+*/
+extern "C" __device__ int meminfo_from_new_udf_str(void** nb_retval, void* udf_str)
+{
+  // allocate enough room for both the meminfo and udf_string
+  meminfo_and_str* mi_and_str = (meminfo_and_str*)NRT_Allocate(sizeof(meminfo_and_str));
+  if (mi_and_str != NULL) {
+    auto mi_ptr        = &(mi_and_str->mi);
+    udf_string* st_ptr = &(mi_and_str->st);
+
+    // We pass a null size here because the udf_string actually exists on the stack
+    // and tracks the size of the string data that it points to.
+    NRT_MemInfo_init(mi_ptr, st_ptr, NULL, udf_str_dtor, NULL);
```

Comment:
It's unclear which is the size without going and referring to the `NRT_MemInfo_init` prototype. Passing a null size is also weird, because it's a `size_t` and not a pointer.

```suggestion
    size_t size = 0;
    void* dtor_info = nullptr;
    NRT_MemInfo_init(mi_ptr, st_ptr, size, udf_str_dtor, dtor_info);
```